src/app/ReadClient.cpp

```diff
@@ -757,6 +802,24 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
                 attributePath.mListOp = ConcreteDataAttributePath::ListOperation::ReplaceAll;
             }
 
+            if (attributePath ==
+                ConcreteDataAttributePath(kRootEndpointId, Clusters::IcdManagement::Id,
+                                          Clusters::IcdManagement::Attributes::OperatingMode::Id))
+            {
+                PeerType peerType;
+                if (CHIP_NO_ERROR == ReadICDOperationModeFromAttributeDataIB(dataReader, peerType))
```

Comment:
I think this would be a lot clearer if:

1. We made the reader copy here, so you don't have to wonder whether ReadICDOperationModeFromAttributeDataIB mutates the state of the reader we pass it.
2. ReadICDOperationModeFromAttributeDataIB took a `TLVReader &&` so it's clear at the callsite that we are handing over the reader we pass it and that it _will_ mutate its state.

Reply:
Makes sense, updated